common/djangoapps/philu_commands/management/commands/tests/test_create_certificates_image_command.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import unicode_literals
+
+import mock
+from certificates.tests.factories import GeneratedCertificateFactory
+from django.core.management import call_command
+from django.db.models.signals import post_save
+from factory.django import mute_signals
+from lms.djangoapps.onboarding.tests.factories import UserFactory
+from student.tests.factories import CourseEnrollmentFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class CreateCertificateImage(SharedModuleStoreTestCase):
+    """
+        Tests for `create_certificates_image` command.
+    """
+
+    @mute_signals(post_save)
+    def setUp(self):
+        """
+        This function is responsible for creating user and courses for every test and mocking the function for tests.
+        :return:
+        """
+        super(CreateCertificateImage, self).setUp()
+        self.user = UserFactory()
+        self.course = CourseFactory()
+        self.course.certificates_display_behavior = "early_with_info"
+        patcher = mock.patch(
+            'openedx.features.student_certificates.tasks.task_create_certificate_img_and_upload_to_s3.delay')
+        self.mock_request = patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def test_create_certificates_images_command(self):
+        """
+        This Test case checks the scenario for creating the certificates images of all of the Generated Certificates.
+        """
+        certificate_uuid = self._create_certificate('honor')
+        call_command('create_certificates_image')
+        self.mock_request.assert_called_once_with(verify_uuid=certificate_uuid)
+
+    def test_create_certificates_images_command_with_date_argument(self):
+        """
+        This Test case checks the scenario for creating the certificates images of Generated Certificates after the
+        given date.
+        """
+        certificate_uuid = self._create_certificate('honor')
+        call_command('create_certificates_image', '--after=30/6/2019')
+        self.mock_request.assert_called_once_with(verify_uuid=certificate_uuid)
+
+    def test_create_certificates_images_command_with_uuid_argument(self):
+        """
+        This Test case checks the scenario for creating the certificates images of Generated Certificates for the given
+        UUID.
+        """
+        certificate_uuid = self._create_certificate('honor')
+        call_command('create_certificates_image', '--uuid={}'.format(certificate_uuid))
+        self.mock_request.assert_called_once_with(verify_uuid=certificate_uuid)
+
+    def _create_certificate(self, enrollment_mode):
```

Comment:
Rename method name according to its purpose.

Reply:
Done